### ISSUE TITLE
Redeploying an appsync API disconnects any existing connected resolvers

### ISSUE THREAD

**Issue by hypexr** [serverless-components/graphql#13]
When an appsync api is deployed and then extended with additional graphql components that reference it's `appId` the resolvers they attached are disconnected.

To fix each component that references the `appId` need to be removed and deployed.

**Comment by eahefnawy** [comment]
yeah right now when you do deploy, we try to match the aws environment to exactly what you have defined in the yaml file. So we remove everything that is not defined, even if it's deployed by another component.

I think we need to have some sort of ownership concept where each instance could only deploy/remove the routes they own/created.

**Comment by hypexr** [comment]
@eahefnawy Any updates on this?  I've been using a cloned component in my org that gets around the problem by not removing anything on deploy/delete.  I'd love to switch back to the official component and have some PRs to put in for lambda versioning and provisioned concurrency that will be easier once I don't have to work around this.
